File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3107_minimum-operations-to-make-median-of-array-equal-to-k/solutions/cpp/solution.cpp
#include <vector>
#include <algorithm>
#include <cmath>

class Solution {
public:
    long long minOperationsToMakeMedianK(std::vector<int>& nums, int k) {
        std::sort(nums.begin(), nums.end());
        int n = nums.size();
        int medianIndex = n / 2;
        long long operations = 0;
        
        // If the array has an even number of elements, choose the higher middle element
        if (n % 2 == 0) {
            medianIndex = n / 2;
        } else {
            medianIndex = n / 2;
        }
        
        // Calculate the number of operations needed
        for (int i = 0; i < n; i++) {
            operations += std::abs(nums[i] - k);
        }
        
        return operations;
    }
};

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3107_minimum-operations-to-make-median-of-array-equal-to-k/solutions/java/solution.java
class Solution {
    public long minOperationsToMakeMedianK(int[] nums, int k) {
        int n = nums.length;
        int medianIndex = (n - 1) / 2; // This is the index of the median in 0-based index array
        long totalOps = 0;
        
        // Sort the array to determine the current median's position
        Arrays.sort(nums);
        
        // Calculate the total number of operations required to make the median equal to k
        for (int i = medianIndex; i < n; i++) {
            totalOps += Math.abs(nums[i] - k);
        }
        return totalOps;
    }
}
